src/arweave/composite-client.ts

```diff
@@ -554,62 +565,62 @@ export class ArweaveCompositeClient
     const url = `/${transactionType}/${txId}`;
     let downloadedFromPeer = true;
 
-    const responsePromise = this.txStore
-      .get(txId)
-      .then((tx) => {
+    const responsePromise = (async () => {
```

Comment:
Nice refactor! :clap:

Reply:
more if that async IIF pattern, ugly wrapper but good profits